pkg/executor/api.go

```diff
@@ -148,9 +149,12 @@ func (executor *Executor) getServiceForFunction(ctx context.Context, fn *fv1.Fun
 		respChan: respChan,
 	}
 	resp := <-respChan
-	if resp.err != nil {
-		return "", resp.err
+	if resp.funcSvc != nil && errors.Is(ctx.Err(), context.Canceled) {
+		et := executor.executorTypes[resp.funcSvc.Executor]
```

Comment:
One minor change, so that we dont make call in case of unknown executor.

```
if et,ok  := executor.executorTypes[resp.funcSvc.Executor]; ok {
     et.UnTapService(ctx, crd.CacheKey(resp.funcSvc.Function), resp.funcSvc.Address)
} else {
     logger.Error("unknown executor type received in function service")
}
return "", ferror.MakeError(499, "client leave early in the process of getServiceForFunction")

```

Reply:
This ends up being more complex as there are multiple cases. LMK if you are ok me raising fix.

```go
func (executor *Executor) getServiceForFunction(ctx context.Context, fn *fv1.Function) (string, error) {
	respChan := make(chan *createFuncServiceResponse)
	executor.requestChan <- &createFuncServiceRequest{
		context:  ctx,
		function: fn,
		respChan: respChan,
	}
	resp := <-respChan

	unTapFunc := func(funcSvc *fscache.FuncSvc) {
		et, ok := executor.executorTypes[funcSvc.Executor]
		if ok {
			et.UnTapService(ctx, crd.CacheKey(funcSvc.Function), resp.funcSvc.Address)
		} else {
			executor.logger.Error("unknown executor type received in function service", zap.Any("executor", funcSvc.Executor))
		}
	}
	if errors.Is(ctx.Err(), context.Canceled) {
		if resp.funcSvc != nil {
			unTapFunc(resp.funcSvc)
		}
		return "", ferror.MakeError(499, "client leave early in the process of getServiceForFunction")
	}
	if resp.err != nil {
		if resp.funcSvc != nil {
			unTapFunc(resp.funcSvc)
		}
		return "", resp.err
	}
	return resp.funcSvc.Address, resp.err
}
```

Reply:
Done, and update some code styles.